### ISSUE TITLE
Unable to resolve path when import uses path relative to `baseUrl`

### ISSUE THREAD

**Issue by OliverJAsh** [alexgorbatchev/eslint-import-resolver-typescript#68]
This is a difficult one to describe but here is a failing test: https://github.com/OliverJAsh/eslint-import-resolver-typescript/commit/e26fed290150682d05974aeb69242281b532c9e0

Essentially the issue occurs when we import using a path that is relative to the `baseUrl`, e.g. given:

`tests/withPaths2/tsconfig.json`:
```json
{
  "compilerOptions": {
    "baseUrl": "root/",
    "paths": {
      "other/*": ["../other/*"]
    }
  },
  "files": ["root/foo.ts", "other/bar.ts"]
}
```
`tests/withPaths2/root/foo.ts`:
```ts
// import using tsconfig.json path mapping
import 'other/bar'
```

`tests/withPaths2/other/bar.ts`:
```ts
// import using tsconfig.json `baseUrl`
// This one fails ❌
import 'foo'
```

This fails with the following error:

```
/Users/oliverash/Development/eslint-import-resolver-typescript/tests/withPaths2/other/bar.ts
  2:8  error  Unable to resolve path to module 'foo'  import/no-unresolved
```

This is a valid import—TypeScript understands it so I think this resolver should also understand it.

**Comment by JounQin** [comment]
We're using `tsconfig-paths` to inside, so maybe you can debug if it's a bug of that module first.

**Comment by OliverJAsh** [comment]
I just tested it with `tsconfig-paths` and it seems to work fine:

```js
const lib = require('tsconfig-paths')
const config = lib.loadConfig()
const f = lib.createMatchPath(config.absoluteBaseUrl, config.paths)
const result = f('foo', undefined, undefined, ['.ts', '.tsx', '.d.ts'])
console.log(result)

```

After some debugging it seems like the problem is with this condition:

https://github.com/alexgorbatchev/eslint-import-resolver-typescript/blob/82ef3573fa1258e0de8d8181de57ae7109e35ec5/src/index.ts#L234-L237

Because of this the function returns `undefined`—we never call `matchPatch`:

![image](https://user-images.githubusercontent.com/921609/107637663-66acd980-6c66-11eb-8102-5c3f85f80945.png)

It seems that the problem happens when we have a module that lives outside of the base URL. 

If I remove that early return then my test works as expected. It doesn't seem to break any of the other tests. Do you know why that condition was added?

**Comment by JounQin** [comment]
@teoxoy Any idea?

**Comment by teoxoy** [comment]
It looks like it was added as part of https://github.com/alexgorbatchev/eslint-import-resolver-typescript/commit/ad337809a5c287f795e328cbc7a1eeefdfbe0ad9#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R125-R128. It was most likely a misunderstanding on my part on how `baseUrl` works.

@OliverJAsh could you put up a PR for this and remove those few lines of code?

**Comment by OliverJAsh** [comment]
https://github.com/alexgorbatchev/eslint-import-resolver-typescript/pull/69

**Comment by JounQin** [comment]
released [v2.4.0](https://www.npmjs.com/package/eslint-import-resolver-typescript/v/2.4.0).
